### PR TITLE
Escape package names to support names which include a slash

### PR DESCRIPTION
**Pull request by bn4t**
Package names aren't escaped before being added to the url.
E.g. `orgs/foo/packages/container/foo/bar` instead of `orgs/foo/packages/container/foo%2Fbar`

This causes 404 errors for such packages.

### CODE REVIEW

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @bn4t !

For each of the modified methods, could you please add a godoc comment (since this is a change of behavior) and also a unit test that demonstrates the new path escaping?

For example:

```go
// Note that packageName is escaped for the URL path so that you don't need to.
```

**Comment by bn4t** [comment]
@gmlewis Would it be fine if I just changed the existing tests to use a package name with a slash or do cou want me to write one/multiple new tests for this?

**Comment by gmlewis** [comment]
> @gmlewis Would it be fine if I just changed the existing tests to use a package name with a slash or do you want me to write one/multiple new tests for this?

I think it is totally fine to modify the existing tests.  Thanks.

**Comment by bn4t** [comment]
@gmlewis Okay, pushed it. Thanks for handholding me through this, this really shouldn't have taken so long. My fault for not checking the contributing guidelines in detail.

**Comment by gmlewis** [comment]
> @gmlewis Okay, pushed it. Thanks for handholding me through this, this really shouldn't have taken so long. My fault for not checking the contributing guidelines in detail.

No problem, we do have a lot of automation in this repo that is not immediately obvious.

**Review by gmlewis — APPROVED** [review]
Thank you, @bn4t !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
